internal/recover/recover.go

```diff
@@ -112,14 +382,21 @@ func ValidateMemberChanges(oldMembers []cluster.DqliteMember, newMembers []clust
 // CreateRecoveryTarball writes a tarball of filesystem.DatabaseDir to
 // filesystem.StateDir.
 // go-dqlite's info.yaml is excluded from the tarball.
+// The new cluster configuration is included as `recovery.yaml`.
 // This function returns the path to the tarball.
-func CreateRecoveryTarball(filesystem *sys.OS) (string, error) {
+func createRecoveryTarball(filesystem *sys.OS, members []cluster.DqliteMember) (string, error) {
 	tarballPath := path.Join(filesystem.StateDir, "recovery_db.tar.gz")
+	recoveryYamlPath := path.Join(filesystem.DatabaseDir, "recovery.yaml")
+
+	err := writeYaml(recoveryYamlPath, members)
```

Comment:
Right before this write to `recovery.yaml`, we write the same information to `cluster.yaml`, so why do we need both files?

Reply:
The schema are different:
```yaml
# cluster.yaml
- ID: 3297041220608546238
  Address: 127.0.0.1:8001
  Role: 0
...
```
```yaml
# recovery.yaml
- id: 3297041220608546238
  address: 127.0.0.1:8001
  role: voter
  name: c1
...
```
Since the incoming addresses may have changed, we have no way of knowing which truststore entry to update with it's new address unless we have a mapping of `name -> address`.